src/jackdaw/serdes/avro.clj

```diff
@@ -600,7 +606,8 @@
     :as   registry-config}
    {:keys [avro/schema
            avro/coercion-cache
-           key?]
+           key?
+           deserializer-properties]
```

Comment:
I think this is trying to do too many things at once. If we want a SerDes that takes a reader schema, we should probably pass it two args with the writer schema required and the reader schema optional. If we just want a deserializer that returns a SerDes that can only be used to consume, I think this should be a separate top level function. (Without a writer schema, it no longer returns a "serde".) How about `deser`?